### PR TITLE
PUBDEV-6208: SE Model deletion can lead to removal of shared vecs (rel-xu port)

### PR DESCRIPTION
**Pull request by seb-h2o**
https://0xdata.atlassian.net/browse/PUBDEV-6208

applying fix from https://github.com/h2oai/h2o-3/pull/3350 to rel-xu

### CODE REVIEW

**Comment by angela0xdata** [comment]
@seb-h2o are you trying to get this into 3.22.1.6 release?

File: h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java [hunk 1]
```diff
@@ -176,7 +176,14 @@ private Frame prepareTrainingLevelOneFrame(StackedEnsembleModel.StackedEnsembleP
         }
       }
 
-      return prepareLevelOneFrame(levelOneKey, baseModels.toArray(new Model[0]), baseModelPredictions.toArray(new Frame[0]), _model._parms.train());
+      Frame levelOneFrame = prepareLevelOneFrame(levelOneKey, baseModels.toArray(new Model[0]), baseModelPredictions.toArray(new Frame[0]), _model._parms.train());
+      if (_parms._keep_levelone_frame) {
+        levelOneFrame.write_lock(_job);
+        levelOneFrame = levelOneFrame.deepCopy(levelOneFrame._key.toString());
+        levelOneFrame.unlock(_job);
+        DKV.put(levelOneFrame);
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java]: Perhaps a little nicer is to use 

   levelOneFrame.update(_job)
   levelOneFrame.unlock(_job)

Comment (seb-h2o) [reply h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java]: yep! changed.

**Comment by seb-h2o** [comment]
@angela0xdata : I was trying to indeed, but I still have some failing tests on Jenkins, so it won't be in 3.22.1.6

**Pull request merged**